### ISSUE TITLE
MAC doesn't detect Mozilla VPN on Fedora 35

### ISSUE THREAD

**Issue by dannycolin** [mozilla/multi-account-containers#2246]
- Multi-Account Containers Version: 8.0.4
- Operating System + Version: Fedora 35
- Firefox Version: 97.0a1 (Nightly)
- Other installed Add-ons + Version + Enabled/Disabled-Status:
```
Add-ons Search Detection	extension	2.0.0	true	addons-search-detection@mozilla.com
Amazon.co.uk	extension	1.9	true	amazon@search.mozilla.org
Bing	extension	1.3	true	bing@search.mozilla.org
BMO Search	extension	0.1	true	bmo@search.dannycolin.com
ClearURLs	extension	1.21.0	true	{74145f27-f039-47ce-a470-a662b129930a}
Ctrl + Shift + C Should Copy	extension	0.1.0	true	ctrl-shift-c-copy@jeffersonscher.com
Dark Reader	extension	4.9.43	true	addon@darkreader.org
DuckDuckGo	extension	1.1	true	ddg@search.mozilla.org
eBay	extension	1.3	true	ebay@search.mozilla.org
Firefox Multi-Account Containers	extension	8.0.4	true	@testpilot-containers
GitHub Search	extension	0.1	true	github@search.dannycolin.com
Google	extension	1.1	true	google@search.mozilla.org
Moz Developer Network (MDN) Search	extension	0.1	true	mdn@search.dannycolin.com
Stylus	extension	1.5.21	true	{7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}
SUMO Live Helper	extension	1.5.8	true	sumo_live_helper@mozillabr.org
Support Moz (SUMO) Search	extension	0.1.1	true	sumo@search.dannycolin.com
Twitter One-Click Search	extension	0.1.1	true	twitter@search.dannycolin.com
uBlock Origin	extension	1.39.2	true	uBlock0@raymondhill.net
Undo Close Tab	extension	7.4.0	true	{4853d046-c5a3-436b-bc36-220fd935ee1d}
Unpaywall	extension	3.98	true	{f209234a-76f0-4735-9920-eb62507a54cd}
Wikipedia (en)	extension	1.1	true	wikipedia@search.mozilla.org
About Sync	extension	0.0.52	false	aboutsync@mhammond.github.com
Code Radio	extension	0.2	false	{e03ca490-b89a-4505-afed-79224d5272d9}
Custom Scrollbars	extension	3.2	false	customscrollbars@computerwhiz
Firefox Color	extension	2.1.7	false	FirefoxColor@mozilla.com
Firefox Relay	extension	2.0.0	false	private-relay@firefox.com
GNOME Shell integration	extension	10.1	false	chrome-gnome-shell@gnome.org
HTTPS Everywhere	extension	2021.7.13	false	https-everywhere@eff.org
Privacy Badger	extension	2021.11.23.1	false	jid1-MnnxcxisBPnSXQ@jetpack
Side View	extension	0.4.6423	false	side-view@mozilla.org
Temporary Containers	extension	1.9.2	false	{c607c8df-14a7-4f28-894f-29e8722976af}
Tree Style Tab	extension	3.8.17	false	treestyletab@piro.sakura.ne.jp
uMatrix	extension	1.4.4	false	uMatrix@raymondhill.net
User-Agent Switcher and Manager	extension	0.4.7.1	false	{a6c4a591-f1b2-4f03-b3ff-767e5bedf4e7}
Violentmonkey	extension	2.13.0	false	{aecec67f-0d10-4fa7-b7c7-609a2db280cf}
Vue.js devtools	extension	5.3.4	false	{5caff8cc-3d2e-4110-a88a-003cc85b3858}
webhint	extension	2.3.5	false	{e748cb59-4901-4bea-b74a-1d8dab98e3c7}
```

### Actual behavior

MAC doesn't detect that Mozilla VPN 2.7.0 is running on my computer so I can't configure the location server for a container.

![image](https://user-images.githubusercontent.com/7339076/146803648-650b719d-45dc-4769-b810-9503f8b02bdc.png)

### Expected behavior

MAC should detect Mozilla VPN and let me configure it in a container

### Steps to reproduce

1. Install Mozilla VPN from [the official GitHub repository](https://github.com/mozilla-mobile/mozilla-vpn-client/suites/4714575065/artifacts/129458689)
2. Install MAC from AMO
3. Open Manage Containers in MAC
4. Choose a container

### Notes

**[SOLVED]** I enabled ipv6 and MAC now detects Mozilla VPN.

**Comment by rpl** [comment]
@dannycolin Firefox on Fedora 35 is using a flatpak package, right?

If that is the case as I'm assuming, then this is likely a duplicate of [Bug 1738441](https://bugzilla.mozilla.org/show_bug.cgi?id=1738441) (which is currently pending on my needinfo request, I'm going to give it a look as soon as possible and then update that bugzilla issue).

**Comment by dannycolin** [comment]
> @dannycolin Firefox on Fedora 35 is using a flatpak package, right?
> 
> If that is the case as I'm assuming, then this is likely a duplicate of [Bug 1738441](https://bugzilla.mozilla.org/show_bug.cgi?id=1738441) (which is currently pending on my needinfo request, I'm going to give it a look as soon as possible and then update that bugzilla issue).

No, I use the tarball version from Mozilla.

**Comment by billyzs** [comment]
I'm having this issue as well on Arch with natively installed firefox. The VPN daemon is enabled via systemd and running, the VPN app can establish connections, but MAC extension does not detect that the VPN is available. I've tried reinstalling the extension, making sure the permission to talk to other applications is enabled, but it didn't help. Is this a problem with the extension or the VPN? If it's with VPN, where should I file this bug to? MozillaVPN's github issues page seem to be a mirror of an internal JIRA instance.

```
% pacman -Qi firefox
Name            : firefox
Version         : 95.0.2-1.1

% pacman -Qi mozillavpn
Name            : mozillavpn
Version         : 2.6.0-1
```

MAC version:
```
Firefox Multi-Account Containers	extension	8.0.4	true	@testpilot-containers
```

```
% systemctl status mozillavpn
● mozillavpn.service - MozillaVPN D-Bus service
     Loaded: loaded (/usr/lib/systemd/system/mozillavpn.service; enabled; vendor preset: disabled)
     Active: active (running) since Sat 2022-01-08 23:41:33 CET; 1h 51min ago
   Main PID: 886 (mozillavpn)
      Tasks: 10 (limit: 18868)
     Memory: 59.3M
        CPU: 4.061s
     CGroup: /system.slice/mozillavpn.service
             └─886 /usr/bin/mozillavpn linuxdaemon
```

```
 dbus-inspect -system org.mozilla.vpn.dbus 
/
  org.mozilla.vpn.dbus
    Methods
      version() → (arg_0 String)
      activate(jsonConfig String) → (arg_0 Bool)
      deactivate() → (arg_0 Bool)
      status() → (jsonStatus String)
      runningApps() → (arg_0 String)
      firewallApp(appName String, state String) → (arg_0 Bool)
      firewallPid(rootpid Int32, state String) → (arg_0 Bool)
      firewallClear() → (arg_0 Bool)
      getLogs() → (logs String)
      cleanupLogs() → ()
    Signals
      connected(hopindex Int32)
      disconnected(hopindex Int32)
  org.freedesktop.DBus.Properties
    Methods
      Get(interface_name String, property_name String) → (value Variant)
      Set(interface_name String, property_name String, value Variant) → ()
      @org.qtproject.QtDBus.QtTypeName.Out0 = QVariantMap
      GetAll(interface_name String) → (values Dict{String, Variant})
    Signals
      @org.qtproject.QtDBus.QtTypeName.Out1 = QVariantMap
      PropertiesChanged(interface_name String, changed_properties Dict{String, Variant}, invalidated_properties Array[String])
  org.freedesktop.DBus.Introspectable
    Methods
      Introspect() → (xml_data String)
  org.freedesktop.DBus.Peer
    Methods
      Ping() → ()
      GetMachineId() → (machine_uuid String)
```

**Comment by dannycolin** [comment]
@billyzs what's your DE/WM? 

Also, I opened the issue here since it's MAC that should communicated with the VPN via the Native Messaging API. However, feel free to also open one on the Mozilla VPN github repository. The devs are monitoring it. If you do so make sure to mention this issue.

**Comment by billyzs** [comment]
I'm using kde's native Wayland session.

**Comment by billyzs** [comment]
I'm using kde's native Wayland session.

**Comment by Cimbali** [comment]
This may be a duplicate of the (later) #2315. I see the specfile for Firefox on Fedora35 uses `echo "ac_add_options --libdir=\"%{_libdir}\""` − not sure about arch builds.

To confirm, you could go to the `about:buildconfig` page and check that the configure options contain `--libdir=/usr/lib64`. You should also see occasional `ExtensionError: No such native application mozillavpn ExtensionUtils.jsm:58:5` error messages in the Browser Console.

The proper fix is to get a version of mozilla VPN that contains mozilla-mobile/mozilla-vpn-client@998149e72f4a4850ee5aef0ee1fda3a051ff96c8 (so currently master, as the latest release to date, 2.7.1, does not contain it) and build with `qmake-qt5 mozillavpn.pro LIBDIR=/usr/lib64`

A (temporary) workaround is to move the VPN file from `/usr/lib/mozilla/native-messaging-hosts/mozillavpn.json` to `/usr/lib64/mozilla/native-messaging-hosts/mozillavpn.json`

**Comment by dannycolin** [comment]
> This may be a duplicate of the (later) https://github.com/mozilla/multi-account-containers/issues/2315. I see the specfile for Firefox on Fedora35 uses echo "ac_add_options --libdir=\"%{_libdir}\"" − not sure about arch builds.

I use the Firefox (Nightly) build from Mozilla 

> To confirm, you could go to the about:buildconfig page and check that the configure options contain --libdir=/usr/lib64. You should also see occasional ExtensionError: No such native application mozillavpn ExtensionUtils.jsm:58:5 error messages in the Browser Console.

`--libdir` isn't even present on the Mozilla build...

```
MOZ_AUTOMATION=1 MOZILLA_OFFICIAL=1 --enable-update-channel=nightly MOZBUILD_STATE_PATH=/builds/worker/.mozbuild MOZ_FETCHES_DIR=/builds/worker/fetches ENABLE_CLANG_PLUGIN=1 MOZ_STDCXX_COMPAT=1 --enable-profile-use=cross --with-pgo-profile-path=/builds/worker/fetches/merged.profdata --with-pgo-jarlog=/builds/worker/fetches/en-US.log MOZ_LTO=cross MOZ_NO_PIE_COMPAT=1 --enable-js-shell --enable-rust-simd --enable-default-toolkit=cairo-gtk3-wayland --with-mozilla-api-keyfile=/builds/mozilla-desktop-geoloc-api.key --with-google-location-service-api-keyfile=/builds/gls-gapi.data --with-google-safebrowsing-api-keyfile=/builds/sb-gapi.data --with-branding=browser/branding/nightly
```

> A (temporary) workaround is to move the VPN file from /usr/lib/mozilla/native-messaging-hosts/mozillavpn.json to /usr/lib64/mozilla/native-messaging-hosts/mozillavpn.json

Even if `--libdir` wasn't present, I tried this workaround just in case but as I was suspecting it didn't resolve the issue.

**Comment by dannycolin** [comment]
Enabling ipv6 solved my issue.
